utils/configs.go

```diff
@@ -0,0 +1,262 @@
+func ValidateDBOptions(binary string, ctx *cli.Context) (*DBConfigs, error) {
+	var storeDriver string
+	var storeURL string
+	var storeURLs string
+	etcdURLs := ctx.String("etcd")
+	consulURLs := ctx.String("consul")
+
+	if etcdURLs != "" && consulURLs != "" {
+		return nil, fmt.Errorf("ambiguous %s db endpoints, both etcd and consul specified: etcd: %s, consul: %s", binary, etcdURLs, consulURLs)
+	} else if etcdURLs == "" && consulURLs == "" {
+		return nil, fmt.Errorf("unknown %s db endpoints", binary)
+	} else if etcdURLs != "" {
+		storeDriver = "etcd"
+		storeURLs = etcdURLs
+	} else {
+		storeDriver = "consul"
+		storeURLs = consulURLs
+	}
+	for _, endpoint := range strings.Split(storeURLs, ",") {
+		_, err := url.Parse(endpoint)
+		if err != nil {
+			return nil, fmt.Errorf("invalid %s %v endpoint: %v", binary, storeDriver, endpoint)
+		}
+		// TODO: support multi-endpoints
+		storeURL = endpoint
+		logrus.Infof("Using %s state db endpoints: %v: %v", binary, storeDriver, storeURL)
+		break
+	}
+
+	return &DBConfigs{
+		StoreDriver: storeDriver,
+		StoreURL:    storeURL,
+	}, nil
+}
+
+// ValidateNetworkOptions returns error if network options are not valid
+func ValidateNetworkOptions(binary string, ctx *cli.Context) (*NetworkConfigs, error) {
+	// 1. validate and set plugin mode
+	pluginMode := strings.ToLower(ctx.String("mode"))
+	switch pluginMode {
+	case core.Docker, core.Kubernetes, core.SwarmMode, core.Test:
+		logrus.Infof("Using %s mode: %v", binary, pluginMode)
+	case "":
+		return nil, fmt.Errorf("%s mode is not set", binary)
+	default:
+		return nil, fmt.Errorf("unknown %s mode: %v", binary, pluginMode)
+	}
+
+	// 2. validate and set network mode
+	networkMode := strings.ToLower(ctx.String("netmode"))
+	switch networkMode {
+	case "vlan", "vxlan":
+		logrus.Infof("Using %s network mode: %v", binary, networkMode)
+	case "":
+		return nil, fmt.Errorf("%s network mode is not set", binary)
+	default:
+		return nil, fmt.Errorf("unknown %s network mode: %v", binary, networkMode)
+	}
+
+	// 3. validate forwarding mode
+	forwardMode := strings.ToLower(ctx.String("fwdmode"))
+	if forwardMode == "" {
+		return nil, fmt.Errorf("unknown %s forwarding mode: %v", binary, forwardMode)
+	} else if forwardMode != "bridge" && forwardMode != "routing" {
+		return nil, fmt.Errorf("%s forwarding mode is not set", binary)
+	} else if networkMode == "vxlan" && forwardMode == "bridge" {
+		return nil, fmt.Errorf("invalid %s forwarding mode: %q (network mode: %q)", binary, forwardMode, networkMode)
+	}
+	// vxlan/vlan+routing, vlan+bridge are valid combinations
+	logrus.Infof("Using %s forwarding mode: %v", binary, forwardMode)
+	return &NetworkConfigs{
+		Mode:        pluginMode,
+		NetworkMode: networkMode,
+		ForwardMode: forwardMode,
+	}, nil
+}
+
+// FlattenFlags concatenate slices of flags into one slice
+func FlattenFlags(flagSlices ...[]cli.Flag) []cli.Flag {
+	flags := flagSlices[0]
```

Comment:
i don't think golang allows more than one slice parameter ( it only works for the last parameter)

Reply:
and here, i thought i was on to something . . ok fine, though now I wonder if this could be ...[]interface{} arg and be a general purpose

Anyway . . . I'm back to first comment, start range at 0 so that you can handle 0 length input?